src/assisted_test_infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -309,8 +311,10 @@ def attach_test_disk(self, node_name, disk_size, bootable=False, persistent=Fals
         # We don't use `vd` virtio disks because libvirt overwrites our aliases if we do so, coming up with
         # its own `virtio-<num>` aliases instead. Those aliases allow us to identify disks created by this
         # function when we perform `detach_all_test_disks` for cleanup.
-        target_dev = f"sd{self._get_available_scsi_identifier(node)}"
-        disk_alias = f"{self.TEST_DISKS_PREFIX}-{target_dev}"
+        dev = {"scsi": "sd", "sata": "sd", "usb": "sd", "virtio": "vd"}
+        target_dev = f"{dev[bus]}{self._get_available_identifier(node, None)}"
+        # Allow to create multiple devices when node shutdown required (sata)
+        disk_alias = f"{self.TEST_DISKS_PREFIX}{str(random.randint(0, 10000))}-{bus}-{target_dev}"
```

Comment:
Please explain what is the random for

Reply:
There are two ways to add: 
- VIR_DOMAIN_AFFECT_LIVE  when support hotplugin (apply when execute on running node )
- VIR_DOMAIN_AFFECT_CURRENT - for SATA we must shutdown /start the node.

Not mandatory but if  the change does not return new target_dev and we want to run the command multiple time it is going to prevent duplicate disk_alias which bubble libvirt exceptions...
for example:
attach_disk_persistent(master, ODF_DISK_SIZE, bus="sata")  --> success
attach_disk_persistent(master, ODF_DISK_SIZE, bus="sata")  --> fail  without restart but indicate that  target sdb already exists  ......

Reply:
so we can try catch this node.attachDeviceFlags(attach_template, attach_flags).....
and if it raise  exception we can increment the target_dev .... not critical .
if you dont like it... i can remove